### PR TITLE
add event loop group

### PR DESCRIPTION
**Pull request by tvd12**
1. [x] Add RoundRoBin
2. [x] Add EzyFuture.
3. [x] Add EventLoopGroup
4. [x] Replace single thread by event loop group
5. [x] Integration test

### CODE REVIEW

File: util/EzyRoundRobin.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            }
+        }
+
+        public T get()
+        {
+            T last = default(T);
+            lock(queue) {
+                last = queue.poll();
+                queue.offer(last);
+            }
```
Comment (vu-luong) [inline util/EzyRoundRobin.cs:37]: Can we use `queue.peek()` here?

Comment (vu-luong) [reply util/EzyRoundRobin.cs:37]: Ah nvm, I got it, it takes the first item and adds it back to the end of queue (round-robin)

File: concurrent/EzyEventLoopGroup.cs [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+                EzyEventLoopEvent runEvent,
+                long delayTime,
+                long period
+            )
+            {
+                this.period = period;
+                this.runEvent = runEvent;
+                this.nextFireTime = DateTime.Now.AddMilliseconds(
+                    delayTime <= 0 ? 0 : period
+                );
```
Comment (vu-luong) [inline concurrent/EzyEventLoopGroup.cs:366]: Could you please elaborate on the logic to use `delayTime` here?

File: socket/EzySocketWriter.cs [hunk 3]
```diff
@@ -11,18 +11,53 @@ protected override void update()
...
+                if (!active)
+                {
+                    return false;
+                }
+                EzyPacket packet = packetQueue.poll();
+                if (packet == null)
+                {
+                    return true;
+                }
+                int writtenBytes = writeToSocket(packet);
```
Comment (vu-luong) [inline socket/EzySocketWriter.cs:46]: Could you explain why we don't need `writeToSocketAsync` but we need `readSocketDataAsync`?

Comment (tvd12) [reply socket/EzySocketWriter.cs:46]: Because `write` is not blocking or only blocking very short time so we don't need to use `write async`

**Review by vu-luong — APPROVED** [review]
LGTM!

**Pull request merged**